dynamicio/mixins/with_s3.py

```diff
@@ -114,8 +183,23 @@ def _read_from_s3_path_prefix(self) -> pd.DataFrame:
         full_path_prefix = utils.resolve_template(f"s3://{bucket}/{path_prefix}", self.options)
 
         # The `no_disk_space` option should be used only when reading a subset of columns from S3
-        if self.options.pop("no_disk_space", False) and file_type == "parquet":
-            return self._read_parquet_file(full_path_prefix, self.schema, **self.options)
+        if self.options.pop("no_disk_space", False):
+            no_disk_space_rv = None
+            if file_type == "parquet":
+                no_disk_space_rv = self._read_parquet_file(full_path_prefix, self.schema, **self.options)
+            elif file_type == "hdf":
+                dfs = []
+                for fobj in self._iter_s3_files(
+                    full_path_prefix,
+                    file_ext=".h5",
+                    max_memory_use=1024**3,  # 1 gib
+                ):
+                    dfs.append(HDFIo().load(fobj))
+                df = pd.concat(dfs, ignore_index=True)
+                columns = [column for column in df.columns.to_list() if column in self.schema.keys()]
+                no_disk_space_rv = df[columns]
+            if no_disk_space_rv is not None:
+                return no_disk_space_rv
```

Comment:
Why not rewrite as: 
```python
        if self.options.pop("no_disk_space", False):
            if file_type == "parquet":
                return self._read_parquet_file(full_path_prefix, self.schema, **self.options)
            elif file_type == "hdf":
                dfs = []
                for fobj in self._iter_s3_files(full_path_prefix, file_ext=".h5", max_memory_use=1024**3):
                    dfs.append(HDFIo().load(fobj))
                df = pd.concat(dfs, ignore_index=True)
                columns = [column for column in df.columns.to_list() if column in self.schema.keys()]
                return df[columns]
            
```

Reply:
I would also urge you to encapsulate this: 
```python
                dfs = []
                for fobj in self._iter_s3_files(full_path_prefix, file_ext=".h5", max_memory_use=1024**3):
                    dfs.append(HDFIo().load(fobj))
                df = pd.concat(dfs, ignore_index=True)
                columns = [column for column in df.columns.to_list() if column in self.schema.keys()]
                return df[columns]
``` 
into a dedicated method. But I see your point on doing this later for csv and json files too!